chronos/asyncfutures2.nim

```diff
@@ -236,6 +236,10 @@ template complete*(future: Future[void]) =
   ## Completes a void ``future``.
   complete(future, getSrcLocation())
 
+template complete*(future: Future[void], val: untyped) =
```

Comment:
hm, can we avoid this one somehow? or at least prevent it from being called with an actual value - ie like this, it looks like `complete(fut, 42)` will compile: it needs a check to ensure that val isn't a .. value

Reply:
Besides a beautiful `static: assert type(x) is void`, don't see a better way right now. But maybe this PR needs a new approach altogether

Reply:
I *think* #401 nails it but can you take a look?

Reply:
Yup, a lot better
Played a bit with it, couldn't make it fail horribly